tmt/steps/report/junit.py

```diff
@@ -76,34 +109,8 @@ def go(self, *args: Any, **kwargs: Any) -> None:
         """ Read executed tests and write junit """
         super().go()
 
-        import_junit_xml()
-        assert junit_xml
-
-        suite = junit_xml.TestSuite(self.step.plan.name)
-        for result in self.step.plan.execute.results():
-            try:
-                main_log = self.step.plan.execute.read(result.log[0])
-            except (IndexError, AttributeError):
-                main_log = None
-            case = junit_xml.TestCase(
-                result.name,
-                classname=None,
-                elapsed_sec=duration_to_seconds(result.duration),
-                stdout=main_log
-                )
-            # Map tmt OUTCOME to JUnit states
-            if result.result == "error":
-                case.add_error_info(result.result)
-            elif result.result == "fail":
-                case.add_failure_info(result.result)
-            elif result.result == "info":
-                case.add_skipped_info(result.result)
-            elif result.result == "warn":
-                case.add_error_info(result.result)
-            # Passed state is the default
-            suite.test_cases.append(case)
-
-        assert self.workdir is not None
+        suite = make_junit_xml(self)
+
         f_path = self.opt("file", os.path.join(self.workdir, DEFAULT_NAME))
```

Comment:
mypy complains: `tmt/steps/report/junit.py:119: error: Argument 1 to "join" has incompatible type "Optional[str]"; expected "Union[str, PathLike[str]]"  [arg-type]`

That's correct, `self.workdir` is `Optional[str]` and can be `None`. I assume workdir cannot be `None` anymore, at this point in workflow (@lukaszachy am I right?), adding `assert self.workdir is not None` would narrow the type.

Reply:
Two more reports:

```
tmt/steps/report/junit.py:123: error: Item "None" of "Optional[Module]" has no attribute "to_xml_report_file"  [union-attr]
tmt/steps/report/junit.py:126: error: Item "None" of "Optional[Module]" has no attribute "TestSuite"  [union-attr]
```

Similar situation, `junit_xml` is `Optional[types.Module]`, adding `assert junit_xml is not None` would narrow the type - we *know* `junit_xml` is not `None` anymore, but mypy cannot infer that on its own.